src/TypeScript/TsConfig/TsConfigProvider.cs

```diff
@@ -0,0 +1,67 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.TypeScript.EslintBridgeClient;
+
+namespace SonarLint.VisualStudio.TypeScript.TsConfig
+{
+    internal interface ITsConfigProvider
+    {
+        /// <summary>
+        /// Returns the tsconfig file for the given source file, or null if an applicable tsconfig could not be found.
+        /// </summary>
+        Task<string> GetConfigForFile(string sourceFilePath, IEslintBridgeClient eslintBridgeClient, CancellationToken cancellationToken);
```

Comment:
As we discussed offline, `IEslintBridgeClient` is not MEF exportable and hence we need to accept it as a param.
I'm working on the TypescriptAnalyzer and I think I will create a MEF-exportable `ITypeScriptEslintBridgeClient` that this class could import. This is a workaround for now.